src/core/Scene.ts

```diff
@@ -1204,6 +1208,27 @@ export class Scene extends ImperativeBase {
 			this.__elementParentSize = parentSize
 		}
 	}
+
+	#caster = new Raycaster()
+
+	#handleClick(ev: MouseEvent) {
+		const pointer = new Vector2()
+
+		pointer.x = (ev.clientX / this.clientWidth) * 2 - 1
+		pointer.y = -(ev.clientY / this.clientHeight) * 2 + 1
+
+		// Setup raycaster
+		this.#caster.setFromCamera(pointer, this.__threeCamera)
+		const intersections = this.#caster.intersectObject(this.three, true)
+		if (intersections.length == 0) return
+
+		// Create custom event with detail of the exact XYZ position of the click
+		const newEvent = new CustomEvent('click', {
```

Comment:
We can use MouseEvent to emulate the real event and its properties https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event